lib/yapay_ex.ex

```diff
@@ -3,16 +3,32 @@ defmodule YapayEx do
   Documentation for `YapayEx`.
   """
 
+  use Application
+
+  alias Finch.Response
   alias YapayEx.{Order, Request}
 
+  def start(_type, _args) do
+    children = [
+      YapayEx.Request.child_spec()
+    ]
+
+    opts = [strategy: :one_for_one, name: YapayEx.Supervisor]
+    Supervisor.start_link(children, opts)
+  end
+
   def create_order(%Order{} = order) do
     endpoint = "/transactions/payment"
 
     case Request.post(endpoint, order) do
-      {:ok, %{status: status, body: body}} when status in 200..299 ->
+      {:ok, %Response{status: status, body: body}} when status in 200..299 ->
+        body =
+          body
+          |> Jason.decode!()
```

Comment:
Nitpicking but you could do that to avoid the one line pipe operator `|>`: 
```suggestion
        body = Jason.decode!(body)
```

Generally, we try to avoid that on Elixir 😄